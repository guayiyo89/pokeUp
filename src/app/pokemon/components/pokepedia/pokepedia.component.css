h2 {
    color: rgb(38, 97, 207);
    font-weight: bolder;
    text-shadow: 2px 2px 0px rgb(255, 217, 0);
    transform: rotate(-3deg);
}

h4 {
    color: rgb(38, 97, 207);
    font-weight: bolder;
    background-color: rgb(255, 217, 0);
    text-align: center;
}

.title{
    padding-top: 8px;
    text-align: center;
}