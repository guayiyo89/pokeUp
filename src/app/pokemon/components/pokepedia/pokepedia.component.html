<div class="container">
    <div class="row">
        <div class="col-12 title">
            <h2>Pokepedia!</h2>
            <p>En esta seccion puedes encontrar un listado de informacion de lugares, regiones, clasificaci√≥n, etc. </p>
        </div>

        <div class="col-4">
            <h4>Habitats</h4>
            <div *ngFor="let item of habitats">
                {{item.name}}
            </div>
        </div>
        <div class="col-4">
            <h4>Regiones</h4>
            <div *ngFor="let item of regiones">
                {{item.name}}
            </div>
        </div>
        <div class="col-4">
            <h4>Tipos</h4>
            <div *ngFor="let item of tipos">
                {{item.name}}
            </div>
        </div>
    </div>
</div>

