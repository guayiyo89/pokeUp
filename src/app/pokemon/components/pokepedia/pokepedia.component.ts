import { Component, OnInit } from '@angular/core';
import { Base } from '../../interfaces/base';
import { PokemonService } from '../../services/pokemon.service';

@Component({
  selector: 'app-pokepedia',
  templateUrl: './pokepedia.component.html',
  styleUrls: ['./pokepedia.component.css']
})
export class PokepediaComponent implements OnInit {
  habitats: Base[] = []
  regiones: Base[] = []
  tipos: Base[] = []

  constructor(private pokeSvc: PokemonService) { }

  ngOnInit(): void {
    this.tipos = this.pokeSvc.dataTypes;
    this.regiones = this.pokeSvc.dataRegion;
    this.habitats = this.pokeSvc.dataHabitat;
  }

}
