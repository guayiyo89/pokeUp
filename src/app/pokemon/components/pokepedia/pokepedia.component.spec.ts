import { ComponentFixture, TestBed } from '@angular/core/testing';

import { PokepediaComponent } from './pokepedia.component';

describe('PokepediaComponent', () => {
  let component: PokepediaComponent;
  let fixture: ComponentFixture<PokepediaComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ PokepediaComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(PokepediaComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
