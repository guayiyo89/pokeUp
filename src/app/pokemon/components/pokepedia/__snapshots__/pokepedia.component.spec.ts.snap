// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokepediaComponent When initializing #should do match with snapshot 1`] = `
"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12 title\\"><h2>Pokepedia!</h2><p>En esta seccion puedes encontrar un listado de informacion de lugares, regiones, clasificaci√≥n, etc. </p></div><div class=\\"col-4\\"><h4>Habitats</h4><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--><div> cave </div><div> forest </div></div><div class=\\"col-4\\"><h4>Regiones</h4><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--><div> chile </div><div> san marino </div></div><div class=\\"col-4\\"><h4>Tipos</h4><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--><div> grandes </div><div> chicos </div></div></div></div>"
`;
